### PR TITLE
Migrate demo page tests to Playwright

### LINKED ISSUE #44553: [Flaky Test] content should load, making the post dirty
<!-- __META_DATA__:{} -->
**Flaky test detected. This is an auto-generated issue by GitHub Actions. Please do NOT edit this manually.**

## Test title
content should load, making the post dirty

## Test path
`/test/e2e/specs/editor/local/demo.spec.js`

## Errors
<!-- __TEST_RESULTS_LIST__ -->
<!-- __TEST_RESULT__ --><time datetime="2022-11-09T18:15:38.168Z"><code>[2022-11-09T18:15:38.168Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/3430531250"><code>fix/constrast-checker-appears-unexpectdly-on-block-without-text-or-link-color</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-03-02T13:05:15.234Z"><code>[2023-03-02T13:05:15.234Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/4313891258"><code>try/preload-navigation-requests</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-03-07T12:01:46.069Z"><code>[2023-03-07T12:01:46.069Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/4353563197"><code>try/dedicated-block-controls-filter</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-03-22T11:42:54.715Z"><code>[2023-03-22T11:42:54.715Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/4489165478"><code>fix/48586-2</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-05-11T04:10:36.513Z"><code>[2023-05-11T04:10:36.513Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/4943674422"><code>behaviors-ui</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-05-12T11:51:39.804Z"><code>[2023-05-12T11:51:39.804Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/4958265493"><code>behaviors-ui</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-05-12T13:04:28.485Z"><code>[2023-05-12T13:04:28.485Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/4958782893"><code>behaviors-ui</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-05-12T20:57:12.039Z"><code>[2023-05-12T20:57:12.039Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/4962657921"><code>add/preview-extensibility</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-05-13T10:07:53.974Z"><code>[2023-05-13T10:07:53.974Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/4966195997"><code>behaviors-ui</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-06-09T14:22:40.976Z"><code>[2023-06-09T14:22:40.976Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5222245159"><code>add/custom-attribute-sources</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-06-10T18:13:04.597Z"><code>[2023-06-10T18:13:04.597Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5231153027"><code>fix/issue-51396-useDisabled-param</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-06-12T11:42:27.745Z"><code>[2023-06-12T11:42:27.745Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5242456031"><code>add/custom-attribute-sources</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-06-23T09:34:32.182Z"><code>[2023-06-23T09:34:32.182Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5354436273"><code>try/aggressive-tinymce-deprecation</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-06-23T10:02:21.385Z"><code>[2023-06-23T10:02:21.385Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5354436273"><code>try/aggressive-tinymce-deprecation</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-06-28T16:45:30.138Z"><code>[2023-06-28T16:45:30.138Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5403385144"><code>update/image-block-deprecations-cleanup</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-07-06T06:07:08.874Z"><code>[2023-07-06T06:07:08.874Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5471957891"><code>try/list-for-patterns</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-07-10T16:20:33.292Z"><code>[2023-07-10T16:20:33.292Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5510156357"><code>fix/navigation-title-in-details</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-07-26T09:10:13.179Z"><code>[2023-07-26T09:10:13.179Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5666320768"><code>add/preview-extensibility</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-07-28T15:39:26.655Z"><code>[2023-07-28T15:39:26.655Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5693422590"><code>fix/apply-default-attributes-on-migration</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-07-28T16:13:24.990Z"><code>[2023-07-28T16:13:24.990Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5693725917"><code>fix/apply-default-attributes-on-migration</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-08-09T20:00:27.957Z"><code>[2023-08-09T20:00:27.957Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5813250847"><code>add/preview-extensibility</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-08-25T12:13:44.193Z"><code>[2023-08-25T12:13:44.193Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5975298932"><code>fix/innerblocks-allowed-blocks-change-with-different-sizes</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-08-30T12:54:18.136Z"><code>[2023-08-30T12:54:18.136Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6024828150"><code>update/auto-inserting-blocks-remove-toggle-if-block-is-present-elsewhere</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-08-31T14:11:38.909Z"><code>[2023-08-31T14:11:38.909Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6036439195"><code>feat/async-block-loading</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-09-04T17:56:01.797Z"><code>[2023-09-04T17:56:01.797Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6076195296"><code>fix/innerblocks-allowed-blocks-change-with-different-sizes</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-09-08T07:47:35.412Z"><code>[2023-09-08T07:47:35.412Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6116679281"><code>tooltip/deprecate-position</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-09-14T14:58:53.059Z"><code>[2023-09-14T14:58:53.059Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6186862966"><code>try/editor-dequeue-block-assets</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-09-22T17:03:00.956Z"><code>[2023-09-22T17:03:00.956Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6276716510"><code>enhancement/update-button-component-deprecate-ispressed</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-09-29T13:06:55.674Z"><code>[2023-09-29T13:06:55.674Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6351638051"><code>try/standardise-link-to-data-mapping</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><time datetime="2023-10-11T05:31:57.045Z"><code>[2023-10-11T05:31:57.045Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6478101822"><code>try/core-data-entity-revisions</code></a>.<!-- /__TEST_RESULT__ -->
<br/>
<!-- __TEST_RESULT__ --><details>
<summary>
	<time datetime="2023-11-07T22:52:09.050Z"><code>[2023-11-07T22:52:09.050Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6790573202"><code>experiment/show-styles-inherited-values</code></a>.
</summary>

```
Error: expect(received).toBe(expected) // Object.is equality

Expected: true
Received: false
    at /home/runner/work/gutenberg/gutenberg/test/e2e/specs/editor/local/demo.spec.js:21:21
```
</details><!-- /__TEST_RESULT__ -->
<!-- /__TEST_RESULTS_LIST__ -->

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.
Closes #44553.

PR migrates demo page tests to Playwright.

## Why?
The test was very flaky due to CORS errors.

## How?
* It looks like Playwright doesn't fail tests due to CORS errors; it just stops loading the image. The test doesn't do any assertions with images; so far so good.
* The original test was mocking Vimeo ember request. The embed block isn't used for assertion; skipping this mock doesn't affect performance.
* I migrated only two assertions that the test was performing.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/local/demo.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/local/demo.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+				.dispatch( 'core/preferences' )
+				.set( 'core/edit-post', 'fullscreenMode', false );
+
+			return true;
+		} );
+
+		const isDirty = await page.evaluate( () => {
+			return window.wp.data.select( 'core/editor' ).isEditedPostDirty();
+		} );
+		expect( isDirty ).toBeTruthy();
```
Comment (WunderBart) [inline test/e2e/specs/editor/local/demo.spec.js]: nit: I think it's worth to be explicit here:

```suggestion
		expect( isDirty ).toBe( true );
```

File: test/e2e/specs/editor/local/demo.spec.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			window.wp.data
+				.dispatch( 'core/preferences' )
+				.set( 'core/edit-post', 'welcomeGuide', false );
+
+			window.wp.data
+				.dispatch( 'core/preferences' )
+				.set( 'core/edit-post', 'fullscreenMode', false );
+
+			return true;
+		} );
```
Comment (WunderBart) [inline test/e2e/specs/editor/local/demo.spec.js:25]: Would it make sense rather to extend the `createNewPost` util? For example:

```js
export async function createNewPost( {
	postType,
	title,
	content,
	excerpt,
	customQuery,
	showWelcomeGuide = false,
} = {} ) {
	const query = addQueryArgs( '', {
		post_type: postType,
		post_title: title,
		content,
		excerpt,
		...customQuery,
	} ).slice( 1 );
	// ...
```

And then in the test:

```js
await admin.createNewPost( {
	customQuery: { 'gutenberg-demo': null },
} );
```

Comment (Mamaduka) [reply test/e2e/specs/editor/local/demo.spec.js:25]: It's probably not worth it for a single use case, and this duplication isn't that massive :)

**Review by kevin940726 — APPROVED** [review]
LGTM 👍

**Pull request merged**